pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -142,3 +142,18 @@ func ValidateResolvedTaskResources(ctx context.Context, params []v1beta1.Param,
 
 	return nil
 }
+
+func ValidateTaskSpecRequestResources(ctx context.Context, taskSpec *v1beta1.TaskSpec) error {
```

Comment:
Is this method only exported so that it can be tested from `package taskrun_test`?

If so, this is generally not a best practice. It both means the method will be unnecessarily exported to external users, who may depend on its behavior not changing. The reason the test is in `package taskrun_test` is to be able to test only exported behaviors, so perhaps we should change it to `package taskrun` and test unexported methods/behaviors.